pkg/upload/react-native-android.go

```diff
@@ -76,7 +76,11 @@ func ProcessReactNativeAndroid(apiKey string, appManifestPath string, bundlePath
 		}
 
 		if projectRoot == "" {
-			projectRoot = path
+			if filepath.Base(path) == "android" {
+				projectRoot = filepath.Join(path, "..")
+			} else {
+				projectRoot = path
+			}
```

Comment:
I think it would be easier to understand (and less error-prone if the directory just happened to be called "android" for another reason) if we applied this at the same point as we're working out the `buildDirPath` (L37):

```go
		buildDirPath := filepath.Join(path, "android", "app", "build")
		rootDirPath := path
		if !utils.FileExists(buildDirPath) {
			buildDirPath = filepath.Join(path, "app", "build")
			if utils.FileExists(buildDirPath) {
				rootDirPath = filepath.Join(path, "..")
			} else if bundlePath == "" || sourceMapPath == "" {
				return fmt.Errorf("unable to find bundle files or source maps in within " + path)
			}
		}
```

Then you just use this value for project root if not already set:

```suggestion
			projectRoot = rootDirPath
```

Reply:
I think you need to amend this check now - it can just assign `projectRoot = rootDirPath`?

Reply:
You are correct. That is my fault I forgot I had the other check further down